app/jobs/appointment_notification/worker.rb

```diff
@@ -86,7 +86,7 @@ def send_message(notification, communication_type, recipient_number:)
         context: context
       )
     when "imo"
-      notification_service.send_notification(notification.patient, notification.localized_message)
+      notification_service.send_notification(notification)
```

Comment:
It looks like IMO will return a `post_id` for messages (see page 6 of the [docs](https://docs.google.com/document/d/1zaTouxdfGg4IqrkCk59KAP905vwynON5Up2ckUax8Mg/).  We can capture that in ImoDeliveryDetails similiar to how we grab the `sid` from Twilio and store it.

This will let us better track IMO notifications going forward, and will more closely match the Twilio flow.  I think the detail workflow will then go from `sent` directly to `read` once the callback comes back in, at least until IMO provides more callback statuses.

Reply:
I completely refactored this per your feedback. The `sent` to `read` behavior isn't really a change, but the way we look up the record is different.